blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/IncomingMetricMetadataAnalyzer.java

```diff
@@ -98,15 +98,17 @@ private IncomingMetricException checkMeta(Locator locator, String key, String in
 
         IncomingMetricException typeProblem = checkMeta(metric.getLocator(), MetricMetadata.TYPE.name().toLowerCase(),
                 metric.getDataType().toString());
-        IncomingMetricException unitProblem = checkMeta(metric.getLocator(), MetricMetadata.UNIT.name().toLowerCase(),
-                metric.getUnit());
+        if (!Configuration.getInstance().getBooleanProperty(CoreConfig.USE_ES_FOR_UNITS)) {
+            IncomingMetricException unitProblem = checkMeta(metric.getLocator(), MetricMetadata.UNIT.name().toLowerCase(),
+                    metric.getUnit());
+            if (unitProblem != null) {
+                problems.add(unitProblem);
+            }
+        }
 
```

Comment:
We need to change the skipping of metadata here instead of in scanMetrics. I will add the change to PR #428